flow.py

```diff
@@ -179,7 +180,7 @@ def acknowledge(self, packet):
         print("Flow", self.flow_id, ": sending acknowledgement packet",
             packet.packet_id)
         a_packet = make_acknowledgement_packet(packet.packet_id, packet.src,
-            packet.dest, packet.packet_size, acknowledge_id, new_id)
+            packet.dest, packet.packet_size, self.num_packets)
```

Comment:
doesn't this function pass in the flow id, but not the packet size or the number of packets?